Treat namespace as required only when necessary

brianphillips opened the pull request:
If a `subject` is specified when registering a new schema, the `namespace` is not used at all and since the schema registry itself doesn't require a namespace, it seems appropriate that this library shouldn't require it either.

Nevon commented:
The `Schema` type should be updated in that case to indicate that `namespace` is optional.

brianphillips commented:
> The `Schema` type should be updated in that case to indicate that `namespace` is optional. And there should be a test verifying that this works as expected.

@Nevon - I have added some tests (although I can't get the existing test suite to run on my local environment, it hangs when exec'ing `docker-compose` it seems).

Regarding the `Schema` type, I was already having to cast to `any` to get my deserialized schema to be accepted because the type expects there to be `toBuffer`, `fromBuffer` and `isValid` methods to exist on the object (which is never the case when deserializing from an `.avsc` file or if you using a schema defined literally in the code or even when retrieving the schema from the registry). I have added a `RawSchema` type as well as some very light validation when reading an `.avsc` file with relevant tests.

Let me know what you think

Nevon commented:
LGTM! Thanks for the contribution!
